src/api/db/models/Camera.ts

```diff
@@ -380,6 +381,61 @@ export class CameraModel {
       throw new InternalServerError(err as string);
     }
   }
+
+  static async deleteCameraTask(
+    input: gql.DeleteCameraInput,
+    context: Pick<Context, 'user' | 'config'>,
+  ): Promise<HydratedDocument<TaskSchema>> {
+    try {
+      console.log('CameraModel.deleteCameraTask - input: ', input);
+      return await TaskModel.create(
+        {
+          type: 'DeleteCamera',
+          projectId: context.user['curr_project'],
+          user: context.user.sub,
+          config: input,
+        },
+        context,
+      );
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  // NOTE: this method is called by the async task handler
+  static async deleteCamera(
```

Comment:
Thinking about moving essentially all of these calls to the corresponding camera task handler, since it seems to be just a series of calls to different models